pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -321,8 +321,13 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
           .validateSegment(segmentMetadata, tempSegmentDir);
 
       // Zk operations
-      completeZkOperations(enableParallelPushProtection, headers, tempEncryptedFile, provider, segmentMetadata,
-          segmentName, zkDownloadUri);
+      if (crypterClassHeader.equalsIgnoreCase(DefaultPinotCrypter.class.getName())) {
```

Comment:
Would like to reduce the number of 'if', if not fully eliminate them.

Reply:
+1
Let us at least add a TODO here if we are in a hurry to ship the fix